pkg/sdk/context.go

```diff
@@ -0,0 +1,45 @@
+package sdk
+
+import (
+	"fmt"
+	"github.com/lunarway/shuttle/pkg/config"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"path"
+)
+
+type ShuttleContext struct {
+	Variables                 config.DynamicYaml `yaml:"vars"` //temporarily include a dynamic representation of the variables here so the go-based plans can use this for templating so we're backwards compatible with the existing templates (for the time being)
+	ProjectPath               string
+	LocalPlanPath             string
+	LocalShuttleDirectoryPath string
+	TempDirectoryPath         string
+}
+
+func LoadShuttleContext(projectPath, localPlanPath string) (ShuttleContext, error) {
+
+	yamlFile, err := LoadShuttleYaml(projectPath)
+	if err != nil {
+		return ShuttleContext{}, err
+	}
+	var result ShuttleContext
+	err = yaml.Unmarshal(yamlFile, &result)
+	if err != nil {
+		return ShuttleContext{}, fmt.Errorf("Failed to parse shuttle configuration. \n\nMake sure your 'shuttle.yaml' is valid. %w", err)
+	}
+
+	result.ProjectPath = projectPath
+	result.LocalShuttleDirectoryPath = path.Join(result.ProjectPath, ".shuttle")
```

Comment:
If we use a local shuttle plan, this is not true, is it?

```yaml
plan: .
```

Reply:
From what I can see the local shuttle folder always points to `projectpath/.shuttle`.
Even when using a local plan the temp files will get written to `projectpath/.shuttle/temp`.

Reply:
Yes, temporary files are always written to `projectPath/.shuttle/temp` but the plan path can be `projectPath` when setting `plan: .` in `shuttle.yaml`. I think at least... 🤔

Reply:
yes, the plan path is written to the `LocalPlanPath` property, two lines below, and I think that is being used to access the plan

Reply:
Ah OK. I might misunderstand the names of these variables. (I've done that before I think).